src/libs/antares/study/parts/short-term-storage/properties.cpp

```diff
@@ -113,6 +142,26 @@ bool Properties::loadKey(const IniFile::Property* p)
 
 bool Properties::validate()
 {
+    if (!injectionCapacity.has_value() || injectionCapacity < 0)
+    {
+        logs.error() << "Property injectionnominalcapacity not provided or negative, needed "
+            << "for short term storage";
+        return false;
+    }
+    if (!withdrawalCapacity.has_value() || withdrawalCapacity < 0)
+    {
+        logs.error() << "Property withdrawalnominalcapacity not provided or negative, needed "
+            << "for short term storage";
+        return false;
+    }
+    if (!capacity.has_value() || capacity < 0)
+    {
+        logs.error() << "Property reservoircapacity not provided or negative, needed "
+            << "for short term storage";
+        return false;
+    }
```

Comment:
> Order of evaluation of any part of any expression, including order of evaluation of function arguments is unspecified (with some exceptions listed below). The compiler can evaluate operands and other subexpressions in any order, and may choose another order when the same expression is evaluated again. 

https://en.cppreference.com/w/cpp/language/eval_order

Reply:
In other words, there is no guarantee that `f` will be executed first in expression `f() || g()`.

There is no choice but to have to `if`